### PR TITLE
Refactor argsort tests.

### PR DESCRIPTION
**Pull request by takagi**
This PR make `argsort` tests clearer in terms of parameterizing tests for `cupy.argsort` and `cupy.ndarray.argsort` before supporting sort multi rank arrays.

### CODE REVIEW

File: tests/cupy_tests/sorting_tests/test_sort.py [hunk 1]
```diff
@@ -128,63 +128,55 @@ def test_lexsort_unsupported_dtype(self, dtype):
             return cupy.lexsort(a)
 
 
+def _argsort(xp, external, a):
```
Comment (okuta) [inline tests/cupy_tests/sorting_tests/test_sort.py]: How about using instance method?

```
def argsort(self, xp, a):
    if self.external:
        return xp.argsort(a)
    else:
        return a.argsort()
```

Comment (takagi) [inline tests/cupy_tests/sorting_tests/test_sort.py]: Fixed!

**Comment by takagi** [comment]
This PR is not needed to backport to v1 because cupy.argsort is introduced at v2.0.0.a1.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**